### DISCUSSION TITLE
Observer for Pivot Model not firing

### DISCUSSION THREAD

**Discussion by Restingo** [GeneaLabs/laravel-model-caching#507] [General]
**Describe the bug**
I've got two models, lets name them `Company` and `Contact`. I've also got a pivot model `CompanyContact`. For the pivot model I've created an observer that is called `CompanyContactObserver`.

Now if model caching is enabled via `.env` file the `CompanyContactObserver` doesn't fire.

**Eloquent Query**
Please provide the complete eloquent query that caused the bug, for example:
```php
$company->contacts()->sync($contacts);
```

**Stack Trace**
None.

**Environment**
 - PHP: 8.1.28
 - OS: Ubuntu 22.04.4 LTS
 - Laravel: 10.44.0
 - Model Caching: 0.13.9

**Additional context**
If model caching is disabled the observer fires instantly.

**Comment by mikebronner** [comment]
@Restingo thank you for reporting this. If you can provide a PR with a failing test, or even the fix, that would help out tremendously.

**Comment by Restingo** [comment]
@mikebronner thanks for your fast response. Would it also be sufficient to commit the models and some test controller? I don't know in which file I need to put the test case :(

**Comment by mikebronner** [comment]
@Restingo Hey, in that case I would recommend just setting up a test repo that demonstrates the problem (comment the code where the problem happens). That will help write the test.
